frontend/packages/vsphere-plugin/src/components/VSphereOperatorStatuses.tsx

```diff
@@ -0,0 +1,173 @@
+  level: number;
+};
+
+const getWorstIconState = (states: OperatorHealthType[]): OperatorHealthType['icon'] => {
+  let worst = states[0];
+  states.forEach((state) => {
+    if (worst.level < state.level) {
+      worst = state;
+    }
+  });
+
+  return worst.icon;
+};
+
+const getOperatorHealth = async (t: TFunction, name: string): Promise<OperatorHealthType> => {
+  try {
+    const operator = await k8sGet<ClusterOperator>({ model: ClusterOperatorModel, name });
+
+    const progressing = getCondition(operator, 'Progressing')?.status as BooleanString | undefined;
+    const available = getCondition(operator, 'Available')?.status as BooleanString | undefined;
+    const degraded = getCondition(operator, 'Degraded')?.status as BooleanString | undefined;
+
+    if (progressing === 'True') {
+      return {
+        message: t('vsphere-plugin~Progressing'),
+        icon: <InProgressIcon />,
+        level: OperatorHealthLevel.Progressing,
+      };
+    }
+
+    if (degraded === 'True') {
+      return {
+        message: t('vsphere-plugin~Degraded'),
+        icon: <ExclamationCircleIcon color={errorColor.value} />,
+        level: OperatorHealthLevel.Degraded,
+      };
+    }
+
+    if (available === 'True') {
+      return {
+        message: t('vsphere-plugin~Healthy'),
+        icon: <CheckCircleIcon color={okColor.value} />,
+        level: OperatorHealthLevel.Healthy,
+      };
+    }
+
+    return {
+      message: t('vsphere-plugin~Unknown'),
+      icon: <UnknownIcon />,
+      level: OperatorHealthLevel.Unknown,
+    };
+  } catch (e) {
+    // eslint-disable-next-line no-console
+    console.error('Error getting operator status: ', name, e);
+    return {
+      message: t('vsphere-plugin~Error'),
+      icon: <ExclamationCircleIcon color={errorColor.value} />,
+      level: OperatorHealthLevel.Error,
+    };
+  }
+};
+
+export const VSphereOperatorStatuses: React.FC = () => {
+  const { t } = useTranslation();
+  const [isExpanded, setIsExpanded] = React.useState(false);
+
+  const initialHealth = {
+    message: t('vsphere-plugin~Pending'),
+    icon: <InProgressIcon />,
+    level: OperatorHealthLevel.Unknown,
+  };
+  const [kubeControllerManager, setKubeControllerManager] = React.useState<OperatorHealthType>(
+    initialHealth,
+  );
+  const [kubeApiServer, setKubeApiServer] = React.useState<OperatorHealthType>(initialHealth);
+  const [storage, setStorage] = React.useState<OperatorHealthType>(initialHealth);
+
+  const [pollingTimmer, setPollingTimmer] = React.useState<number>(0);
+
+  React.useEffect(() => {
+    const doItAsync = async () => {
+      setKubeControllerManager(await getOperatorHealth(t, 'kube-controller-manager'));
+      setKubeApiServer(await getOperatorHealth(t, 'kube-apiserver'));
+      setStorage(await getOperatorHealth(t, 'storage'));
+
+      await delay(DELAY_BEFORE_POLLING_RETRY_MEDIUM);
+      setPollingTimmer(pollingTimmer + 1);
+    };
+
+    doItAsync();
+  }, [pollingTimmer, t]);
```

Comment:
why are you polling instead of using `useK8sWatchResource` ?

Reply:
done

Reply:
I dont see any chage here, did you forget to commit ?

Reply:
old diff, I see it now ! :)